drivers/sensor/bmi160/bmi160.c

```diff
@@ -790,9 +790,11 @@ static inline void bmi160_gyr_channel_get(const struct device *dev,
 					  struct sensor_value *val)
 {
 	struct bmi160_data *data = to_data(dev);
+	void *gyr = data->sample.gyr;
 
-	bmi160_channel_convert(chan, data->scale.gyr,
-			       data->sample.gyr, val);
+	/* Assert below triggers with native_posix. */
+	/*__ASSERT_NO_MSG(IS_PTR_ALIGNED(gyr, uint16_t));*/
```

Comment:
@andyross @dcpleung @peter-mitsis  Again, I am hitting this on native_posix which is x86-32.